Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -642,11 +642,6 @@ public class PBXProjGenerator {
                     )
                     targetFrameworkBuildFiles.append(buildFile)
                 }
-
-                let targetDependency = addObject(
-                    PBXTargetDependency(product: packageDependency)
-                )
-                dependencies.append(targetDependency)
```

Comment:
Last I checked this was required, otherwise it might not compile the package before it's used. Because if it's not linked (which it won't be above if the thing depending on it is a static library) then compilation will fail.

Reply:
@brentleyjones Thank you for your comment.
Does the situation you point out actually occur?
In my opinion, the package will compile correctly because dependencies are specified in `packageProductDependencies`. 

https://github.com/yonaskolb/XcodeGen/blob/080fd34536d1825bbba928a2d8e9595f62d16472/Sources/XcodeGenKit/PBXProjGenerator.swift#L636

Reply:
From when I was testing it @brentleyjones is correct. The strange thing is that if you add a package dependency through Xcode it doesn't actually do this 🤷‍♂ 
Why does `pod install` fail? Maybe that's why Xcode doesn't do it

Reply:
CocoaPods(1.8.4) just doesn't seem to support the package dependency generated by the code I removed.
Sure. I thought that I should fix it, but since Xcode did not output the corresponding part, I sent a PR here first. 😓

If you'd like, should you tell me a little more about when this dependency is needed?
I tried a few, but I couldn't reproduce the problem you mentioned.

Reply:
`Target A`: iOS App
`Target B`: static library
`SwiftPM Package A`

`Target A` Dependencies: `Target B`
`Target B` Dependencies: `SwiftPM Package A`

If the code is removed, then `Target B` can be compiled along side `SwiftPM Package A` and fail, since it depends on `SwiftPM Package A` being fully compiled first.

Reply:
You might be correct that https://github.com/yonaskolb/XcodeGen/blob/080fd34536d1825bbba928a2d8e9595f62d16472/Sources/XcodeGenKit/PBXProjGenerator.swift#L636 handles the situation. I'll have to retest later, if someone doesn't before me.

Reply:
Yeah if you could test these scenarios out @brentleyjones, that would be great!

Reply:
@brentleyjones Thank you for your reply.
I tried the situation you pointed out. So far the compilation hasn't failed with or without the code I removed.

Below are the projects I tried:
https://github.com/tid-kijyun/A

If my understanding is mistaken, please tell me.

Reply:
Also when testing if compilation fails, clean your derived data first

Reply:
I deleted the derived data before each build and tested it.
In my environment, a build error occurred in the following procedure.

```sh
rm -rf /Users/[MyAccount]/Library/Developer/Xcode/DerivedData/*
mint run yonaskolb/xcodegen xcodegen generate
xcodebuild -project A.xcodeproj -scheme B -sdk iphonesimulator
```

Build environment:
* macOS Catalina 10.15.1(19B88)
* Xcode 11.2.1(11B500)
* Apple Swift version 5.1.2 (swiftlang-1100.0.278 clang-1100.0.33.9)
  Target: x86_64-apple-darwin19.0.0
* xcodegen 2.10.1

Reply:
Sorry! There seems to be a problem with the package I'm referencing.
I will test again with different conditions.

Reply:
So `packageProductDependencies` isn't what is making this work for top level targets, it's the linking. Your latest change, where this is only one if we don't link, works.